pkg/controller/operands/kubevirtConfigMap_test.go

```diff
@@ -0,0 +1,212 @@
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsNotFound(err)).To(BeTrue())
+	})
+
+	It("should not remove the configMap during upgrade", func() {
+		req.SetUpgradeMode(true)
+
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kvCmName,
+				Namespace: commonTestUtils.Namespace,
+			},
+			Data: map[string]string{
+				"fakeKey": "fakeValue",
+			},
+		}
+		c := commonTestUtils.InitClient([]runtime.Object{hco, cm})
+		handler := newKubeVirtCmHandler(c, emitter)
+
+		res := handler.ensure(req)
+
+		Expect(res.Err).ToNot(HaveOccurred())
+		Expect(res.UpgradeDone).To(BeTrue())
+
+		Expect(emitter.CheckEvents(expectedEvents)).To(BeFalse())
+
+		resCm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kvCmName,
+				Namespace: commonTestUtils.Namespace,
+			},
+		}
+
+		err := hcoutils.GetRuntimeObject(context.TODO(), c, resCm, req.Logger)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(resCm.Data).ToNot(BeEmpty())
+		Expect(resCm.Data["fakeKey"]).Should(Equal("fakeValue"))
+	})
+
+	It("should return error if failed to read the configMap", func() {
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kvCmName,
+				Namespace: commonTestUtils.Namespace,
+			},
+			Data: map[string]string{
+				"fakeKey": "fakeValue",
+			},
+		}
+		c := commonTestUtils.InitClient([]runtime.Object{hco, cm})
+		fakeError := fmt.Errorf("fake get error")
+		c.InitiateGetErrors(func(key client.ObjectKey) error {
+			if key.Name == kvCmName {
+				return fakeError
+			}
+			return nil
+		})
+
+		handler := newKubeVirtCmHandler(c, emitter)
+
+		res := handler.ensure(req)
+
+		Expect(res.Err).To(HaveOccurred())
+		Expect(res.Err).To(MatchError(fakeError))
+		Expect(res.UpgradeDone).To(BeTrue())
+
+		Expect(emitter.CheckEvents(expectedEvents)).To(BeFalse())
+
+		resCm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kvCmName,
+				Namespace: commonTestUtils.Namespace,
+			},
+		}
+
+		c.InitiateGetErrors(nil) // cancel fake error
+		err := hcoutils.GetRuntimeObject(context.TODO(), c, resCm, req.Logger)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(resCm.Data).ToNot(BeEmpty())
+		Expect(resCm.Data["fakeKey"]).Should(Equal("fakeValue"))
+	})
+
+	It("should return error if failed to delete the configMap", func() {
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      kvCmName,
+				Namespace: commonTestUtils.Namespace,
+			},
+			Data: map[string]string{
+				"fakeKey": "fakeValue",
+			},
+		}
+		c := commonTestUtils.InitClient([]runtime.Object{hco, cm})
+		fakeError := fmt.Errorf("fake delete error")
+		c.InitiateDeleteErrors(func(obj client.Object) error {
+			if unstructed, ok := obj.(runtime.Unstructured); ok {
+				kind := unstructed.GetObjectKind()
+				if kind.GroupVersionKind().Kind == "ConfigMap" && obj.GetName() == kvCmName {
+					return fakeError
+				}
+			}
+			return nil
+		})
```

Comment:
general question: why is `HcoTestClient.deleteError` has a receiver function of `FakeWriteErrorGenerator` that gets `client.Object` and not `client.ObjectKey`?
i.e. why does the delete function needs the whole object, instead just its key?

Reply:
The client.Client interface uses the Writer interface. The Delete function defined like this:

```go
// Delete deletes the given obj from Kubernetes cluster.
Delete(ctx context.Context, obj Object, opts ...DeleteOption) error
```

So this is what the Delete method get. Now, the fake error function should be flixible, to allow different errors, so it get what the Delete function get, so it can play with it.